### PR TITLE
Cloud: show terminal creation state in pre-created split panes

### PR DESCRIPTION
**Pull request by austinywang**
Follow-up to [#12379](https://github.com/manaflow-ai/cmux/pull/12379) and [#12376](https://github.com/manaflow-ai/cmux/issues/12376).

## Problem

The Cloud split UI creates a Bonsplit pane synchronously, then asks the Cloud provider to create and project a terminal asynchronously. If the cmux-tui link is starting, restarting, or temporarily unavailable, the pre-created pane can otherwise remain an anonymous empty pane until a later refresh.

## Change

- Install a temporary Cloud terminal panel immediately in a new Cloud split pane.
- Show localized progress while the remote terminal is being created.
- Show a localized, provider-safe failure state with Retry when creation or projection fails.
- Cancel the in-flight operation when the pending pane closes.
- Reuse the already-created remote terminal after a projection failure, avoiding duplicate terminals on Retry.
- Close an untouched empty Bonsplit pane if its temporary panel cannot be installed.
- Keep Cloud routing authoritative when the provider is unavailable; it cannot silently fall through to a local shell.
- Keep the existing Cloud VM loading view in its own source file and correct the Xcode source-group wiring.

## Trade-offs and scope

The pending panel owns only its observable UI phase; the coordinator is the single owner of the created remote resource and operation generation. The change covers the pane-creation contract in #12376 and deliberately does not duplicate the transport resolver/recovery work in #12362 or the stale workspace-binding work in #12443. Existing attach, portal, and geometry recovery remain separate follow-ups.

## Validation

- `python3 scripts/swift_file_length_budget.py`
- `python3 scripts/localization_catalog.py check` (six catalogs, nine locales, zero parity errors)
- `python3 scripts/normalize-pbxproj.py`
- `./scripts/check-pbxproj.sh`
- `./scripts/lint-pbxproj-test-wiring.sh`
- `git diff --check`
- Focused Swift Testing: `CloudTerminalCreationCoordinatorTests` covers projection-failure resource reuse and pending-pane cancellation.
- GitHub focused E2E run: [test-e2e workflow](https://github.com/manaflow-ai/cmux/actions/workflows/test-e2e.yml) dispatched for `cmuxTests/CloudTerminalCreationCoordinatorTests` on this branch.
- No local Xcode build was run; the tagged remote build is queued through the shared build lock.

## Demo video

Not available yet: the shared GUI verification host was unavailable during this update. The focused behavior test and remote tagged build are the available evidence; no visual result is being claimed until the GUI run completes.

## Review trigger

Please review the latest pushed implementation commits through `3ea553dd18`, plus merge update `a4aef92498` from `origin/main` at `faae269088`. Reviewers should pay particular attention to the temporary-panel lifecycle, cancellation/retry ownership, and the corrected PBX source paths.

## Checklist

- [x] Issue linked and scope limited to Cloud terminal creation panes.
- [x] Regression tests remain in the PR history before the fix commit.
- [x] New user-facing copy uses `String(localized:defaultValue:)` and all nine macOS locale entries.
- [x] Swift file-length and PBX wiring checks pass.
- [ ] Tagged GUI Cloud dogfood and before/after recording (blocked by unavailable verification host).
- [ ] Merge (requires personally verified end-to-end behavior).

### CODE REVIEW

File: Resources/Localizable.xcstrings [hunk 1]
```diff
@@ -6518,6 +6518,360 @@
...
+        "ja": {
+          "stringUnit": {
+            "state": "translated",
+            "value": "cmux-tui がターミナル要求を受け付けませんでした。"
+          }
+        },
+        "de": {
+          "stringUnit": {
+            "state": "translated",
+            "value": "cmux-tui hat die Terminalanfrage nicht akzeptiert."
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Resources/Localizable.xcstrings]: Disposition: disagree. The repository policy in AGENTS.md and skills/cmux-localization requires every new macOS catalog key to include all nine supported locales (en, de, fr, ar, es, zh-Hant, zh-Hans, ko, ja). Removing those entries would fail the localization parity check. I kept the translations and removed internal cmux-tui wording from the user-facing copy.

File: Sources/CloudTerminalCreationCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            guard let self, let panel = self.panel else { return }
+            do {
+                let resource: SurfaceResource
+                if let createdResource = self.createdResource {
+                    resource = createdResource
+                } else {
+                    resource = try await self.create()
+                    guard self.generation == operationGeneration else { return }
+                    self.createdResource = resource
+                    panel.setCreatedResource(resource)
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/CloudTerminalCreationCoordinator.swift]: Fixed in 2a8027420d: CloudTerminalCreationCoordinator is now the sole owner of the created resource. The pending panel cache, setter, coordinator assignment, and test-only assertion were removed; the retry behavior remains covered by the create-count test.

**Comment by austinywang** [comment]
Follow-up to the latest CodeRabbit outside-diff review: the `remoteWorkspaceID` edge case is fixed in `0024d2ea05`. `materializeManualMirrorTerminal` now prefers the workspace carried by the newly-created `SurfaceResource` before falling back to the local binding or daemon focus, so a response without `tab_id` stays in the requested Cloud workspace. Provider responses with an explicit tab placement remain unchanged.

File: Sources/CloudTerminalOverlayCoordinator.swift [hunk 3]
```diff
@@ -26,7 +26,11 @@ final class CloudTerminalOverlayCoordinator {
     /// Creates a coordinator backed by the app's signature dismissal store.
     ///
     /// - Parameter dismissalStore: Repository shared by this native surface owner.
```
[the remark this answers is not in the document]
Comment (austinywang) [reply Sources/CloudTerminalOverlayCoordinator.swift:28]: Fixed in the current uncommitted merge update: the dismissalStore parameter documentation now belongs to init(dismissalStore:), while the no-argument init() has its own summary.

**Comment by austinywang** [comment]
Current review dispositions: the supported macOS localization set remains the nine locales required by AGENTS.md and skills/cmux-localization; the suggested eleven additional catalog locales are not part of the app-supported set, so that finding is intentionally disagreed with. The placement-retry finding is fixed in 3b22dadcb0: the projection receipt now survives retryable resolution outcomes and is returned with the eventual resolved surface. The initializer documentation finding is fixed and resolved in 3b22dadcb0.
